### ISSUE TITLE
Only read from superglobals in controller

### ISSUE THREAD

**Issue by joshuaadickerson** [elkarte/Elkarte#2015]
For the most part, you should only read from the superglobals in the controller. Then, you should be passing that to the model, view, helpers. There are hundreds of cases where we read from them in subs files. That makes it hard to test and really hard to hook.

**Comment by Vekseid** [comment]
Wait till you find the points where they WRITE TO superglobals...

**Comment by Spuds** [comment]
Yup that is still quite the mess, I bet 100's were removed, but its still so ugly.  

The abuse of superglobals, and use of globals to pass between methods are items to be targeted in 1.1  Wish we could have got more of that in 1.0 but there is almost a lifetime supply of them in the code :P

**Comment by emanuele45** [comment]
It's just 600+ or 800+ depending what is counted.
Lots of fun! :tada:

**Comment by Spuds** [comment]
Is that all ... did you forget to count all the places where they are read and then altered in place :tongue: 

Thinking ... should we add a method(s) to the request class to return $_GET[] or $_POST[] vars?

So in the controllers we could just get the instance and then call a method $locals = $req->put(array(of put indexes)) or $locals = $req->get(array(of get indexes)) or along those lines.  Maybe even add a filter param to that.

Also should cleanRequest() be moved to the request class instead of the QueryString file?

**Comment by emanuele45** [comment]
Ping #1655 regarding writing to superglobals
